src/core/nsfs.js

```diff
@@ -93,7 +93,7 @@ async function main(argv = minimist(process.argv.slice(2))) {
         // };
 
         const bs = new BucketSpaceFS({ fs_root });
-        const ns = new NamespaceFS({ fs_root });
+        const ns = new NamespaceFS({ fs_path: fs_root, protocol: '' });
         const object_sdk = new ObjectSDK(null, null, null);
         object_sdk._get_bucketspace = () => bs;
         object_sdk._get_bucket_namespace = async name => ns;
```

Comment:
so the change we need here is a bit different for this to work like before - we need _get_bucket_namespace to be implemented now correctly so that it will create a new namespace based on the bucket name and use `fs_path: fs_root + '/' + bucket.name`. only nuance is that _get_bucket_namespace is called on every op and assumes it won't be heavy, so I would just use a map per bucket name and keep the created namespaces inside it.

Reply:
cool, fixed.